Lib/unittest/test/testmock/testhelpers.py

```diff
@@ -900,6 +901,54 @@ def __getattr__(self, attribute):
         autospec = create_autospec(proxy)
         self.assertFalse(hasattr(autospec, '__name__'))
 
+    def test_spec_inspect_signature(self):
+
+        def myfunc(x, y):
+            pass
+
+        mock = create_autospec(myfunc)
+        assert inspect.getfullargspec(mock) == inspect.getfullargspec(myfunc)
+        mock(1, 2)
+        mock(x=1, y=2)
+        self.assertRaises(TypeError, mock, 1)
+
+        def foo(a: int, b: int=10, *, c:int) -> int:
+            return a + b + c
+
+        mock = create_autospec(foo)
+        assert inspect.getfullargspec(mock) == inspect.getfullargspec(foo)
+        mock(1, 2, c=3)
+        mock(1, c=3)
+        self.assertRaises(TypeError, mock, 1)
+        self.assertRaises(TypeError, mock, 1, 2, 3, c=4)
+
+    def test_spec_inspect_signature_partial(self):
```

Comment:
Can we check also against a combination of `partial` and `partialmethod`?

Reply:
Sorry, by combination do you mean a `partialmethod` taking a `partial` function as an argument? An example would help here.

Reply:
Ideally, I'd prefer to see this split out into its own bug and PR. I'm much less certain about the situation with `partial`, could this be a bug on the partial side that needs fixing there instead? (are partials being created with an incorrect `__signature__` or is this something special about what Mock is doing?)